deltachat-ffi/deltachat.h

```diff
@@ -1096,43 +1096,94 @@ void            dc_set_draft                 (dc_context_t* context, uint32_t ch
  * If needed, the device-chat is created before.
  *
  * Sends the event #DC_EVENT_MSGS_CHANGED on success.
  * To check, if a given chat is a device-chat, see dc_chat_is_device_talk()
  *
  * @memberof dc_context_t
  * @param context The context as created by dc_context_new().
+ * @param label A unique name for the message to add.
+ *     The label is typically not displayed to the user and
+ *     must be created from the characters `A-Z`, `a-z`, `0-9`, `_` or `-`.
  * @param msg Message to be added to the device-chat.
  *     The message appears to the user as an incoming message.
- * @return The ID of the added message.
+ * @return The ID of the added message,
+ *     if the message was already added or skipped before or on errors, 0 is returned.
+ *
+ * Example:
+ * ~~~
+ * dc_msg_t* welcome_msg = dc_msg_new(DC_MSG_TEXT);
+ * dc_msg_set_text(welcome_msg, "great that you give this app a try!");
+ *
+ * dc_msg_t* changelog_msg = dc_msg_new(DC_MSG_TEXT);
+ * dc_msg_set_text(changelog_msg, "we have added 3 new emojis :)");
+ *
+ * if (dc_add_device_msg_once(context, "welcome", welcome_msg)) {
+ *     // do not add the changelog on a new installations -
+ *     // not now and not when this code is executed again
+ *     dc_skip_device_msg(context, "update-123");
+ * } else {
+ *     // welcome message was not added now, this is an oder installation,
+ *     // add a changelog
+ *     dc_add_device_msg_once(context, "update-123", changelog_msg);
+ * }
+ * ~~~
  */
-uint32_t        dc_add_device_msg            (dc_context_t* context, dc_msg_t* msg);
+uint32_t        dc_add_device_msg_once       (dc_context_t* context, const char* label, dc_msg_t* msg);
 
 
 /**
- * Add a message only one time to the device-chat.
- * The device-message is defined by a name.
- * If a message with the same name was added before,
- * the message is not added again.
- * Use dc_add_device_msg() to add device-messages unconditionally.
+ * Add a message to the device-chat unconditionally.
+ * As this skips the test if the message was added before,
+ * normally, you should prefer dc_add_device_msg_once() over this function
  *
  * Sends the event #DC_EVENT_MSGS_CHANGED on success.
  *
  * @memberof dc_context_t
  * @param context The context as created by dc_context_new().
- * @param label A unique name for the message to add.
- *     The label is typically not displayed to the user and
- *     must be created from the characters `A-Z`, `a-z`, `0-9`, `_` or `-`.
  * @param msg Message to be added to the device-chat.
  *     The message appears to the user as an incoming message.
- * @return The ID of the added message,
- *     this might be the id of an older message with the same name.
+ * @return The ID of the added message.
  */
-uint32_t        dc_add_device_msg_once       (dc_context_t* context, const char* label, dc_msg_t* msg);
+uint32_t        dc_add_device_msg_unlabelled(dc_context_t* context, dc_msg_t* msg);
```

Comment:
does this duplication really make sense? I would expect to just not specify a label (eg NULL) with dc_add_device_msg(), to get the unlabelled behaviour.

Reply:
you're right, i did a change.